core/design-system/src/main/kotlin/team/noweekend/core/design/system/core/component/tabbar/item/defaults/NWKNavigationBarItemColors.kt

```diff
@@ -0,0 +1,65 @@
+package team.noweekend.core.design.system.core.component.tabbar.item.defaults
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.graphics.Color
+
+@Immutable
+class NWKNavigationBarItemColors(
+    private val selectedIconColor: Color,
+    private val selectedLabelColor: Color,
+    private val unselectedIconColor: Color,
+    private val unselectedLabelColor: Color,
+    private val disabledIconColor: Color,
+    private val disabledLabelColor: Color,
+) {
+    @Composable
+    internal fun iconColor(selected: Boolean, enabled: Boolean): State<Color> {
+        return rememberUpdatedState(
+            newValue = when {
+                !enabled -> disabledIconColor
+                selected -> selectedIconColor
+                else -> unselectedIconColor
+            },
+        )
+    }
+
+    @Composable
+    internal fun labelColor(selected: Boolean, enabled: Boolean): State<Color> {
+        return rememberUpdatedState(
+            newValue = when {
+                !enabled -> disabledLabelColor
+                selected -> selectedLabelColor
+                else -> unselectedLabelColor
+            },
+        )
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other == null || this::class != other::class) return false
+
+        other as NWKNavigationBarItemColors
+
+        if (selectedIconColor != other.selectedIconColor) return false
+        if (selectedLabelColor != other.selectedLabelColor) return false
+        if (unselectedIconColor != other.unselectedIconColor) return false
+        if (unselectedLabelColor != other.unselectedLabelColor) return false
+        if (disabledIconColor != other.disabledIconColor) return false
+        if (disabledLabelColor != other.disabledLabelColor) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = selectedIconColor.hashCode()
+        result = 31 * result + selectedLabelColor.hashCode()
+        result = 31 * result + unselectedIconColor.hashCode()
```

Comment:
p4: 약간 복잡해서 보여서..! 이렇게도 할 수 있다는데 참고만 해주~

```suggestion
    override fun equals(other: Any?): Boolean =
    this === other || (other is NWKNavigationBarItemColors &&
        selectedIconColor == other.selectedIconColor &&
        selectedLabelColor == other.selectedLabelColor &&
        unselectedIconColor == other.unselectedIconColor &&
        unselectedLabelColor == other.unselectedLabelColor &&
        disabledIconColor == other.disabledIconColor &&
        disabledLabelColor == other.disabledLabelColor
    )

override fun hashCode(): Int =
    arrayOf(
        selectedIconColor,
        selectedLabelColor,
        unselectedIconColor,
        unselectedLabelColor,
        disabledIconColor,
        disabledLabelColor
    ).contentHashCode()
```

Reply:
ㅋㅋㅋㅋㅋ M3 무지성으로 갖다 쓰다가 저렇게 들어가긴 했는데 딱들켰네용 수정할게용
hashCode 저렇게 쓰는건 너무 좋은 것 같아용

Reply:
아 근데 지금 다시 보니까 저게 원래는 NavigationBarItem을 커스텀해서 쓰려고 만든 Color인데 지금 변경된 디자인시스템에서는 기존 Material NavigationBarItem을 그대로 쓸 수 있는 구조라 저 클래스 자체가 필요 없게됨
저 파일 그냥 지워버릴겝!

Reply:
[7ad1e69](https://github.com/mash-up-kr/NoWeekend-Android/pull/34/commits/7ad1e698893dc53bd9f915114cb30eeefd94508c) 수정완료

Reply:
아리가리마시따.